akka-docs-dev/rst/scala/http/common/json-support.rst

```diff
@@ -1,4 +1,32 @@
 JSON Support
 ============
 
-...
\ No newline at end of file
+Akka HTTP's :ref:`marshalling <http-marshalling-scala>` and :ref:`unmarshalling <http-unmarshalling-scala>`
+infrastructure makes it rather easy to seamlessly support specific wire representations of your data objects, like JSON,
+XML or even binary encodings.
+
+For JSON Akka HTTP currently provides support for `spray-json`_ right out of the box through it's
+``akka-http-spray-json`` module.
+
+
+spray-json Support
+------------------
+
+The SprayJsonSupport_ trait provides a ``FromEntityUnmarshaller[T]`` and ``ToEntityMarshaller[T]`` for every type ``T``
+that an implicit ``spray.json.RootJsonReader`` and/or ``spray.json.RootJsonWriter`` (respectively) is available for.
+
+This is how you enable automatic support for (un)marshalling from and to JSON with `spray-json`_:
+
+1. Add a library dependency onto ``"com.typesafe.akka" %% "akka-http-spray-json-experimental" % "1.x"``.
+
+2. ``import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._`` or mix in the
+   ``akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport`` trait.
+
+3. Provide a ``JsonFormat[T]`` for your type and bring it into scope.
+   Check out the `spray-json`_ documentation for more info on how to do this.
+
+Once you have done this (un)marshalling between JSON and your type ``T`` should work nicely and transparently.
+
+
+.. _spray-json: https://github.com/spray/spray-json
+.. _SprayJsonSupport: https://github.com/akka/akka/blob/release-2.3-dev/akka-http-marshallers-scala/akka-http-spray-json/src/main/scala/akka/http/scaladsl/marshallers/sprayjson/SprayJsonSupport.scala
```

Comment:
I think Johannes means using

```
.. _akka.http.scaladsl.Http: @github@/akka-http-core/src/main/scala/akka/http/scaladsl/Http.scala
```

rather than the full link I have above.
I'll convert the links.